.github/actions/prepare-workspace/action.yml

```diff
@@ -0,0 +1,63 @@
+name: Prepare Teleport workspace
+description: Prepares Teleport /workspace filder
+inputs:
+  workload_identity_provider:
+    description: Google Cloud identity provider
+    required: false
+    default: projects/671314886056/locations/global/workloadIdentityPools/github-pool/providers/github-provider
+
+  service_account:
+    description: Google Cloud Storage service account name
+    required: false
+    default: teleport-storage@just-terminus-366813.iam.gserviceaccount.com
+
+  cache_key:
+    description: Cache infix used in cache actions
+    required: false
+    default: ${{ github.workflow }}
+  
+runs:
+  using: "composite"
+  steps:
+    - name: Mark /workspace as git safe.directory
+      shell: bash
+      run: |
+        git config --global --add safe.directory /workspace
+        git config --global --add safe.directory /workspace/webassets
+
+    - name: Authenticate to Google Cloud
```

Comment:
As far as I understood, this log uploader was introduced because GCB has a size limit on logs in the first place. GitHub Actions do not have such limit. The easiest solution would be to just print everything to STDOUT. Log output would not be as handy as it is now, but it would work and comply with the policy.

Another thing is that no secrets here are involved. We do not use any keys or passwords here. Github Actions authenticate via [Workload Identity Federation OAuth flow](https://cloud.google.com/iam/docs/workload-identity-federation) to exchange data with Google. Permissions and restrictions are set on the Google's IAM side.

However, this step may fail in a fork. I guess, a conditional might be required here.

Reply:
@gzigzigzeo We do need to be able to run tests on external contributions from forks. Can this be an optional step?

Reply:
Yes, we can, done. Finally, extracted gcloud login to a separate action.